pkg/controller/managedresources/merger.go

```diff
@@ -36,8 +36,15 @@ func merge(desired, current *unstructured.Unstructured, forceOverwriteLabels boo
 	newObject := current
 	desired.DeepCopyInto(newObject)
 
-	newObject.SetResourceVersion(oldObject.GetResourceVersion())
-	newObject.SetFinalizers(oldObject.GetFinalizers())
+	// keep metadata information of old object to avoid unnecessary update calls
+	if oldMetadataInterface, ok := oldObject.Object["metadata"]; ok {
+		// cast to map to be able to check if metadata is empty
+		if oldMetadataMap, ok := oldMetadataInterface.(map[string]interface{}); ok {
+			if len(oldMetadataMap) > 0 {
+				newObject.Object["metadata"] = oldMetadataMap
```

Comment:
It seems like this was also the case before, but would this approach work if we change something in the metadata, e.g., we add a finalizer to the manifests? It would get overwritten here, or? Do we need to support such cases?

Reply:
Indeed, if you would change metadata other than `labels` or `annotations` it would get overwritten.
I double checked in one of our soils and we don't have any metadata other than `name,namespace,labels,annotations` in all secrets used by managed resources.

As you said, adding a finalizer to the manifest was not supported before and I can't think of a case, where this is actually needed. So I guess, this should be fine, but if we encounter a case where this is a valid requirement, it should be easy to add here :)

Reply:
Yes, sure, alright, thanks for clarifying!